### PR TITLE
ProcedureNotUsedInspection false positive for PropertyLet

### LINKED ISSUE #5430: False positive `Property let accessor '{x}' is not used`
**Rubberduck version information**
```
Version 2.5.0.5384
OS: Microsoft Windows NT 10.0.17134.0, x64
Host Product: Microsoft Office 2016 x64
Host Version: 16.0.4966.1000
Host Executable: EXCEL.EXE
```

**Description**
I have 2 false positive `let` inspections on accessors that are used within the class module

**To Reproduce**
In a class module, I have, among others:
```
Private pCurrentReportRow As Long
Private pPageNumber As Long

Public Property Let PageNumber(ByVal Value As Long)
  pPageNumber = Value
End Property
Public Property Let CurrentReportRow(ByVal Value As Long)
  pCurrentReportRow = Value
End Property

Private Sub Class_Initialize()
  'other initialization statements here
  CurrentReportRow = 1
  PageNumber = 1
  
End Sub
```

**Expected behavior**
Since the `let` accessors are used to assign these values, instead of assigning to their backing fields, they are used and should, therefore, not trigger an inspection.

**Screenshots**
N/A

**Logfile**
[RubberduckLog.txt](https://github.com/rubberduck-vba/Rubberduck/files/4355726/RubberduckLog.txt)

**Additional context**

### LINKED ISSUE #5490: Procedure Not Used Inspection False Positive
**Rubberduck version information**
Version 2.5.0.5422
OS: Microsoft Windows NT 10.0.18362.0, x64
Host Product: Microsoft Office x86
Host Version: 16.0.12730.20236
Host Executable: EXCEL.EXE

**Description**
Subject Inspection flags a property let accessor as not used when it is used in a factory method

**To Reproduce**
FooModel Class (Predeclared)
```
Option Explicit
Implements IFoo

Private Type TModel
    SomeProperty As String
End Type
Private this As TModel
Private Property Get IFoo_SomeProperty() As String
    IFoo_SomeProperty = this.SomeProperty
End Property

Public Function Create(ByVal SomeProperty As String) As IFoo
    With New FooModel
        .SomeProperty = SomeProperty 'Let is used here
        Set Create = .Self
    End With
End Function
Public Property Get Self() As IFoo
    Set Self = Me
End Property
Public Property Let SomeProperty(ByVal value As String)
    this.SomeProperty = value
End Property
```

**Expected behavior**
Inspection would not flag the Let accessor

### PR DESCRIPTION
**Pull request by BZngr**
Closes #5490 
Closes #5430

### CODE REVIEW

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs [hunk 1]
```diff
@@ -20,25 +20,37 @@ namespace Rubberduck.CodeAnalysis.Inspections.Concrete
...
     /// <module name="MyModule" type="Standard Module">
     /// <![CDATA[
     /// Option Explicit
     /// 
+    /// 'No inspection result
     /// Public Sub DoSomething()
     ///     ' macro is attached to a worksheet Shape.
     /// End Sub
+    /// 
+    /// 'Flagged by inspection
```
Comment (retailcoder) [inline Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs:36]: Let's keep the example simples, and add more of them if needed - the procedure that doesn't trigger the inspection should be in an example that doesn't trigger the inspection. The website will list all modules under a collapsible heading for each example:

image

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs [hunk 2]
```diff
@@ -20,25 +20,37 @@ namespace Rubberduck.CodeAnalysis.Inspections.Concrete
...
+    /// Private Sub DoSomethingElse()
+    ///     ' macro is attached to a worksheet Shape.
+    /// End Sub
     /// ]]>
     /// </module>
     /// </example>
     /// <example hasResult="false">
     /// <module name="MyModule" type="Standard Module">
```
Comment (retailcoder) [inline Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs:44]: Nitpick, and not from this PR, but I noticed some examples say `Module1` and others say `MyModule` - I think we should pick one and stick to it... personally I'd prefer `Module1`.

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs [hunk 3]
```diff
@@ -20,25 +20,37 @@ namespace Rubberduck.CodeAnalysis.Inspections.Concrete
...
     /// event handler procedures and interface members that Rubberduck isn't recognizing as such.
+    /// Public procedures of Standard Modules are not flagged by this inspection regardless of
+    /// the presence or absence of user code references.
     /// </remarks>
     /// <example hasResult="true">
     /// <module name="MyModule" type="Standard Module">
     /// <![CDATA[
     /// Option Explicit
     /// 
+    /// 'No inspection result
```
[the remark this answers is not in the document]
Comment (BZngr) [reply Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs]: Redundant in-code comments removed (they will just have to believe the markup :slightly_smiling_face:)

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs (hunk 1 above)
Comment (BZngr) [reply Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs]: Simplified the current examples and added one for Class Module procedure(s) scenario.

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs (hunk 2 above)
Comment (BZngr) [reply Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs]: Modified examples to join the `Module1` camp.

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs [hunk 4]
```diff
@@ -14,32 +15,90 @@ namespace Rubberduck.CodeAnalysis.Inspections.Concrete
...
+    /// ]]>
+    /// </module>
+    /// </example>
+    /// <example hasResult="false">
+    /// <module name="Class1" type="Class Module">
+    /// <![CDATA[
+    /// Option Explicit
+    /// 
     /// Public Sub DoSomething()
-    ///     ' macro is attached to a worksheet Shape.
```
Comment (retailcoder) [inline Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs:42]: That was not redundant or contradicting the markup though? Sorry to sound like I'm dancing back and forth, but comments like this help clarify an example.

What's important is that things like module names and types, and example attributes ("hasresult"), are consistent with the code examples - wasn't the case before this commit, but now the pendulum went all the way to the other end....

Comment (BZngr) [reply Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs:42]: I looked at this a little differently...I did not delete the comment not because I thought it was contradictory to the markup - I agree, it is not.  Rather, I deleted it because I felt it falsely implied to our users that somehow RD 'knew' it was attached to a worksheet shape - and that's _why_ a result is not generated.  The real reason is as stated in the `<remarks>` - that any Public procedure in a Standard Module is ignored - the possibility of attachment to a worksheet shape is one example of why these types of procedures are ignored.

I'm happy to restore the comment - though I might suggest it read something like "macro may be attached to a worksheet Shape or other use"...unless I'm missing something about the original comment (which is also entirely possible).   

Just let me know if my modified version of the comment has merit or you would like it simply restored 'as it was' - I'm good either way.

**Comment by MDoerner** [comment]
Regarding the commit message, please use something more descriptive in the future. In the log, it will be impossible to figure out what this commit is about without actually looking at its contents.

Generally, the first line should be one short sentence describing the change, without a dot at the end. Below, further explanations can be given whenever there is more content than the first line suggests or when an explanation of the reasoning is in order.

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs (hunk 4 above)
Comment (retailcoder) [reply Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs:42]: Good thinking, hadn't realized that implication, makes sense.

We can use module names too; if the module is named `Macros` (std module), that an convey useful information without being too wordy. And we can restate the condition in a comment, e.g. "`' public procedure in a standard module`

**Comment by SmileyFtW** [comment]
Could you add an annotation along the lines: '@EntryPoints (plural) for a module so that procedures in that module that would trigger the inspection be ignored. Could also have procedure scope annotation '@EntryPoint (singular) to only skip the annotated procedure.

**Comment by retailcoder** [comment]
@SmileyFtW that would work, but then it's essentially just aliasing `Ignore ProcedureNotUsed` / `IgnoreModule ProcedureNotUsed`?

**Comment by SmileyFtW** [comment]
@retailcoder You are certainly correct. An annotation indicates a "why" rather than just an "Ignore" with no apparent reason, especially if the module and/or procedure isn't named in a way to indicate the reasoning was my thinking.

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs (hunk 4 above)
Comment (BZngr) [reply Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureNotUsedInspection.cs:42]: Latest update/sync restores a comment within this example
